objects-revisited/main.js

```diff
@@ -0,0 +1,75 @@
+const api = new XMLHttpRequest();
+const business = {
+  opens: '6:00 AM',
+  closes: '7:00 PM',
+  totalEmployees: 2,
+  daysOpen: ['Mon', 'Wed', 'Fri'],
+  employees: {
+    Raymond: {
+      position: 'CEO',
+      daysOfWeekWorking: ['Mon', 'Fri'],
+    },
+    Raul: {
+      position: 'Salesman',
+      daysOfWeekWorking: ['Mon', 'Wed'],
+    },
+  },
+};
+
+function addWeekends() {
+  business.daysOpen.push('Sat', 'Sun');
+
+  for (const employee in business.employees) {
+    business.employees[employee].daysOfWeekWorking.push('Sat', 'Sun');
+  }
+}
+
+function addEmployees() {
+  api.open('GET', 'https://jsonplaceholder.typicode.com/users');
+  api.responseType = 'json';
+  api.addEventListener('load', function () {
+    for (let i = 0; i < 4; i++) {
+      const job = ['Developer', 'Janitor', 'HR', 'Designer'];
+      const employee = api.response[i].name;
+      business.employees[employee] = {
+        position: job[i],
+        daysOfWeekWorking: [],
+        fullTime: null,
+      };
+      const workingDays = business.employees[employee].daysOfWeekWorking;
+
+      // adds random work day
+      for (let e = 0; e < 3; e++) {
+        const random = Math.round(Math.random());
+
+        if (random) {
+          workingDays.push(business.daysOpen[e]);
+        }
+      }
+
+      workingDays.push('Sat', 'Sun');
+
+      // shows if employee is fulltime
+      if (workingDays.length === 5) {
+        business.employees[employee].fullTime = true;
+      } else {
+        business.employees[employee].fullTime = false;
+      }
+
+      // updates total employees
+      business.totalEmployees++;
+    }
+  });
+  api.send();
+}
+
+function deleteEmployee(employeeName) {
+  delete business.employees[employeeName];
+  business.totalEmployees--;
+}
+
+addEventListener('DOMContentLoaded', function () {
+  addWeekends();
+  addEmployees();
+  deleteEmployee('Raymond');
+});
```

Comment:
You also need to add the `fullTime` property to employees inside of here. Right now, you are currently only adding that property to employees that get created after initialization. The employees that exist before aren't getting this property. So "Raul" for example, currently does not have a `fullTime` property

Reply:
<img width="362" alt="image" src="https://github.com/AaronHe09/c0523-code-solutions/assets/79122214/8470e4c5-da6b-472f-9e02-386734e0cc1d">